File: frontend/src/services/carService.js
import api from './api';

const carService = {
    // Lấy tất cả xe
    getAllCars: (page = 0, size = 10) => {
        return api.get(`/cars?page=${page}&size=${size}`);
    },

    // Lấy xe theo ID
    getCarById: (id) => {
        return api.get(`/cars/${id}`);
    },

    // Tìm kiếm xe theo tên
    searchCars: (name) => {
        return api.get(`/cars/search?name=${name}`);
    },

    // Lọc xe theo danh mục
    getCarsByCategory: (catalogId) => {
        return api.get(`/cars/catalog/${catalogId}`);
    },

    // Lọc xe theo khoảng giá
    getCarsByPrice: (minPrice, maxPrice) => {
        return api.get(`/cars/price?min=${minPrice}&max=${maxPrice}`);
    },

    // Lọc xe theo số chỗ
    getCarsBySeat: (socho) => {
        return api.get(`/cars/socho/${socho}`);
    },

    // Lấy xe bán chạy
    getBestSellers: (limit = 6) => {
        return api.get(`/cars/best-sellers?limit=${limit}`);
    },
};

export default carService;

File: oto-shop/src/main/java/com/otostore/oto_shop/Service/CarService.java
package com.otostore.oto_shop.Service;

import com.otostore.oto_shop.Entity.Car;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Optional;

public interface CarService {

    // Lấy tất cả xe
    List<Car> getAllCars();

    // Lấy tất cả xe với phân trang
    Page<Car> getAllCars(Pageable pageable);

    // Lấy xe theo ID (trả về Optional)
    Optional<Car> getCarById(Integer id);

    // Tìm kiếm xe theo tên
    List<Car> searchCarsByName(String name);

    // Lọc xe theo danh mục
    List<Car> getCarsByCatalog(Integer catalogId);

    // Lọc xe theo khoảng giá
    List<Car> getCarsByPriceRange(Double minPrice, Double maxPrice);

    // Lọc xe theo màu sắc
    List<Car> findByMausac(String mausac);

    // Lọc xe theo số chỗ
    List<Car> findBySocho(Integer socho);

    // Lọc xe theo năm sản xuất
    List<Car> findByNamsx(Integer namsx);

    // Thêm xe mới
    Car addCar(Car car);

    // Cập nhật xe
    Car updateCar(Integer id, Car car);

    // Xóa xe
    void deleteCar(Integer id);

    // Upload ảnh
    String uploadImage(Integer carId, MultipartFile file) throws Exception;

    // Lấy xe bán chạy
    List<Car> getBestSellers(int limit);
}